get_state_symbols not working properly after conversion

example: I added the following to test_unit_conversion.py
```
    def test_convert_state_symbols(self, aslanidi_model):
        model = aslanidi_model
        model.get_state_symbols()
        model.units.add_custom_unit('millisecond', [{'prefix': 'milli', 'units': 'second'}])
        time_variable = model.get_free_variable_symbol()
        current_units = model.units.summarise_units(time_variable)
        desired_units = model.units.ureg.millisecond

        model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)       
        model.get_state_symbols()
```
output:
```
====================================================== FAILURES =======================================================
____________________________________ TestUnitConversion.test_convert_state_symbols ____________________________________

self = <tests.test_unit_conversion.TestUnitConversion object at 0x05EE2670>
aslanidi_model = <cellmlmanip.model.Model object at 0x057FF4B0>

    def test_convert_state_symbols(self, aslanidi_model):
        model = aslanidi_model
        model.get_state_symbols()
        model.units.add_custom_unit('millisecond', [{'prefix': 'milli', 'units': 'second'}])
        time_variable = model.get_free_variable_symbol()
        current_units = model.units.summarise_units(time_variable)
        desired_units = model.units.ureg.millisecond

        model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)
>       model.get_state_symbols()

tests\test_unit_conversion.py:759:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cellmlmanip\model.py:290: in get_state_symbols
    state_symbols = [v.args[0] for v in self.get_derivative_symbols()]
cellmlmanip\model.py:274: in get_derivative_symbols
    derivative_symbols = [v for v in self.graph if isinstance(v, sympy.Derivative)]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cellmlmanip.model.Model object at 0x057FF4B0>

    @property
    def graph(self):
        """ A ``networkx.DiGraph`` containing the model equations. """
        # TODO: Set the parameters of the model (parameters rather than use initial values)

        # Return cached graph
        if self._graph is not None:
            return self._graph

        # Store symbols, their attributes and their relationships in a directed graph
        graph = nx.DiGraph()

        equation_count = 0

        # Add a node for every variable in the model, and set additional variable meta data
        for equation in self.equations:
            equation_count += 1

            # Determine LHS.
            lhs = equation.lhs
            if not (lhs.is_Derivative or isinstance(lhs, VariableDummy)):
                raise RuntimeError('DAEs are not supported. All equations must be of form `x = ...` or `dx/dt = ...')

            # Add the lhs symbol of the equation to the graph
            graph.add_node(lhs, equation=equation)

            # Update variable meta data based on the variable's role in the model
            if lhs.is_Derivative:
                # Get the state symbol and update the variable information
                state_symbol = lhs.free_symbols.pop()
                state_symbol.type = 'state'

                # Get the free symbol and update the variable information
                free_symbol = lhs.variables[0]
                free_symbol.type = 'free'
            else:
                lhs.type = None

        # Sanity check: none of the lhs have the same hash
        assert len(graph.nodes) == equation_count

        # Sanity check: all the lhs are unique in meaning (sympy.Dummy: same name != same hash)
        assert len(set([str(x) for x in graph.nodes])) == equation_count

        # Add edges between the nodes
        for equation in self.equations:
            lhs = equation.lhs

            # for each of the symbols or derivatives on the rhs of the equation
            for rhs in self.find_symbols_and_derivatives([equation.rhs]):

                if rhs in graph.nodes:
                    # If the symbol maps to a node in the graph just add the dependency edge
                    graph.add_edge(rhs, lhs)
>               elif rhs.type in ['state', 'free']:
E               AttributeError: 'Derivative' object has no attribute 'type'

cellmlmanip\model.py:482: AttributeError
================================== 1 failed, 127 passed, 1 skipped in 11.17 seconds ===================================
```

--- a/cellmlmanip/model.py
+++ b/cellmlmanip/model.py
@@ -754,7 +754,7 @@ class Model(object):
         :param new_derivative: new variable representing the derivative
         """
         for equation in self.equations:
-            for argument in equation.atoms(sympy.Derivative):
+            for argument in equation.rhs.atoms(sympy.Derivative):
                 if new_derivative['expression'] == argument:
                     # add new equation
                     new_eqn = equation.subs(new_derivative['expression'], new_derivative['variable'])

--- a/cellmlmanip/model.py
+++ b/cellmlmanip/model.py
@@ -754,7 +754,7 @@ class Model(object):
         :param new_derivative: new variable representing the derivative
         """
         for equation in self.equations:
-            for argument in equation.rhs.args:
+            for argument in equation.atoms(sympy.Derivative):
                 if new_derivative['expression'] == argument:
                     # add new equation
                     new_eqn = equation.subs(new_derivative['expression'], new_derivative['variable'])

--- a/tests/cellml_files/repeated_ode_for_conversion_tests.cellml
+++ b/tests/cellml_files/repeated_ode_for_conversion_tests.cellml
@@ -50,15 +50,19 @@
                 <eq/>
                 <ci>x</ci>
                 <apply>
-                    <times/>
+                    <plus/>
+                    <cn cellml:units="mV_per_ms">1</cn>
                     <apply>
-                        <diff/>
-                        <bvar>
-                            <ci>time</ci>
-                        </bvar>
-                        <ci>sv1</ci>
+                        <times/>
+                        <apply>
+                            <diff/>
+                            <bvar>
+                                <ci>time</ci>
+                            </bvar>
+                            <ci>sv1</ci>
+                        </apply>
+                        <cn cellml:units="mV_per_ms">3</cn>
                     </apply>
-                    <cn cellml:units="mV_per_ms">3</cn>
                 </apply>
             </apply>
             <apply>

--- a/tests/test_unit_conversion.py
+++ b/tests/test_unit_conversion.py
@@ -738,8 +738,7 @@ class TestUnitConversion:
             local_model.convert_variable(variable, bad_unit, direction)
 
     def test_convert_same_unit_different_name(self, br_model):
-        """ Tests the Model.convert_variable() function when conversion to current unit under a different name.
-        """
+        """ Tests the Model.convert_variable() function when conversion to current unit under a different name."""
         br_model.units.add_custom_unit('millimolar', [{'units': 'mole', 'prefix': 'milli'},
                                        {'units': 'litre', 'exponent': '-1'}])
         # [{'units': 'mole', 'prefix': 'nano'}, {'units': 'litre', 'exponent': '-3', prefix: 'milli'}])
@@ -747,6 +746,25 @@ class TestUnitConversion:
         variable = br_model.get_symbol_by_ontology_term(shared.OXMETA, "cytosolic_calcium_concentration")
         direction = DataDirectionFlow.INPUT
         assert br_model.convert_variable(variable, unit, direction) == variable
+
+    def test_convert_state_symbols(self, aslanidi_model):
+        """ Tests if get_state_symbols still works (and the graph is rebuilt properly)
+        after after conversion of the time (s to ms).
+
+        The aslanidi_model has derivatives on the rhs of equations that aren't at the top level being replaced.
+        """
+        model = aslanidi_model
+        old_state_symbols = model.get_state_symbols()
+        assert len(old_state_symbols) == 29
+
+        model.units.add_custom_unit('millisecond', [{'prefix': 'milli', 'units': 'second'}])
+        time_variable = model.get_free_variable_symbol()
+        desired_units = model.units.ureg.millisecond
+
+        model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)
+        state_symbols = model.get_state_symbols()
+        assert len(state_symbols) == 29
+        assert old_state_symbols == state_symbols
 
     def test_unique_names(self, silly_names):
         # original state

--- a/tests/test_unit_conversion.py
+++ b/tests/test_unit_conversion.py
@@ -165,8 +165,8 @@ class TestUnitConversion:
         assert symbol_t.units == 'ms'
         assert len(multiode_model.equations) == 3
         assert str(multiode_model.equations[0]) == 'Eq(Derivative(_env_ode$sv1, _environment$time), _1.0)'
-        assert str(multiode_model.equations[1]) == 'Eq(_env_ode$x, ' \
-                                                   '_3.0*Derivative(_env_ode$sv1, _environment$time))'
+        assert str(multiode_model.equations[1]) == \
+            'Eq(_env_ode$x, _1.0 + _3.0*Derivative(_env_ode$sv1, _environment$time))'
         assert str(multiode_model.equations[2]) == 'Eq(Derivative(_env_ode$y, _environment$time), _2.0)'
         return True
 
@@ -441,7 +441,7 @@ class TestUnitConversion:
                 var y: mV {init: 3};
 
                 ode(sv1, time) = 1{mV_per_ms};
-                x = ode(sv1, time) * 3{mV_per_ms};
+                x = 1{mV_per_ms} + ode(sv1, time) * 3{mV_per_ms};
                 ode(y, time) = 2{mV_per_ms};
 
         convert_variable(sv1, volt, DataDirectionFlow.INPUT)
@@ -459,7 +459,7 @@ class TestUnitConversion:
                 sv1 = 1000 * sv1_converted
                 sv1_orig_deriv = 1{mV_per_ms}
                 ode(sv1_converted, time) = 0.001 * sv1_orig_deriv
-                x = 3 * sv1_orig_deriv
+                x = 1{mV_per_ms} + 3 * sv1_orig_deriv
         """
         mV_unit = multiode_model.get_units('mV')
         volt_unit = multiode_model.get_units('volt')
@@ -478,7 +478,7 @@ class TestUnitConversion:
         assert str(multiode_model.equations[2]) == 'Eq(_env_ode$sv1_orig_deriv, _1.0)'
         assert str(multiode_model.equations[3]) == 'Eq(Derivative(_env_ode$sv1_converted, _environment$time), ' \
                                                    '0.001*_env_ode$sv1_orig_deriv)'
-        assert str(multiode_model.equations[4]) == 'Eq(_env_ode$x, _3.0*_env_ode$sv1_orig_deriv)'
+        assert str(multiode_model.equations[4]) == 'Eq(_env_ode$x, _1.0 + _3.0*_env_ode$sv1_orig_deriv)'
 
     def test_multiple_odes_1(self, multiode_model):
         """ Tests the Model.convert_variable function that changes units of given variable.
@@ -493,7 +493,7 @@ class TestUnitConversion:
                 var y: mV {init: 3};
 
                 ode(sv1, time) = 1{mV_per_ms};
-                x = ode(sv1, time) * 3{mV_per_ms};
+                x = 1{mV_per_ms} + ode(sv1, time) * 3{mV_per_ms};
                 ode(y, time) = 2{mV_per_ms};
 
 
@@ -513,11 +513,14 @@ class TestUnitConversion:
                 ode(y, time_converted) = 1000 * y_orig_deriv;
                 sv1_orig_deriv = 1{mV_per_ms}
                 ode(sv1, time_converted) = 1000 * sv1_orig_deriv
-                x = 3 * sv1_orig_deriv
+                x = 1 + 3 * sv1_orig_deriv
         """
         ms_unit = multiode_model.get_units('ms')
         second_unit = multiode_model.get_units('second')
         original_var = multiode_model.get_symbol_by_name('environment$time')
+
+        old_state_symbols = multiode_model.get_state_symbols()
+        assert len(old_state_symbols) == 2
 
         assert self._original_state_multiode_model(multiode_model)
         # test no change in units
@@ -534,7 +537,12 @@ class TestUnitConversion:
         assert str(multiode_model.equations[3]) == 'Eq(_env_ode$y_orig_deriv, _2.0)'
         assert str(multiode_model.equations[4]) == 'Eq(Derivative(_env_ode$y, _environment$time_converted), ' \
                                                    '1000.0*_env_ode$y_orig_deriv)'
-        assert str(multiode_model.equations[5]) == 'Eq(_env_ode$x, _3.0*_env_ode$sv1_orig_deriv)'
+        assert str(multiode_model.equations[5]) == 'Eq(_env_ode$x, _1.0 + _3.0*_env_ode$sv1_orig_deriv)'
+
+        # test the graph forming and get_state_symbols still works
+        state_symbols = multiode_model.get_state_symbols()
+        assert len(state_symbols) == 2
+        assert old_state_symbols == state_symbols
 
     def test_add_output(self, literals_model):
         """ Tests the Model.convert_variable function that changes units of given variable.
@@ -746,25 +754,6 @@ class TestUnitConversion:
         variable = br_model.get_symbol_by_ontology_term(shared.OXMETA, "cytosolic_calcium_concentration")
         direction = DataDirectionFlow.INPUT
         assert br_model.convert_variable(variable, unit, direction) == variable
-
-    def test_convert_state_symbols(self, aslanidi_model):
-        """ Tests if get_state_symbols still works (and the graph is rebuilt properly)
-        after after conversion of the time (s to ms).
-
-        The aslanidi_model has derivatives on the rhs of equations that aren't at the top level being replaced.
-        """
-        model = aslanidi_model
-        old_state_symbols = model.get_state_symbols()
-        assert len(old_state_symbols) == 29
-
-        model.units.add_custom_unit('millisecond', [{'prefix': 'milli', 'units': 'second'}])
-        time_variable = model.get_free_variable_symbol()
-        desired_units = model.units.ureg.millisecond
-
-        model.convert_variable(time_variable, desired_units, DataDirectionFlow.INPUT)
-        state_symbols = model.get_state_symbols()
-        assert len(state_symbols) == 29
-        assert old_state_symbols == state_symbols
 
     def test_unique_names(self, silly_names):
         # original state

